### ISSUE TITLE
I couldn't get the weights-improvement-48.hdf5

### ISSUE THREAD

**Issue by suikammd** [anuragmishracse/caption_generator#22]
\documentclass{article}
\begin{document}
\paragraph{After training, I noticed I just generate weights as follows:}
\paragraph{weights-improvement-01.hdf5}
\paragraph{weights-improvement-02.hdf5}
\paragraph{weights-improvement-03.hdf5}
\paragraph{weights-improvement-04.hdf5}
\paragraph{Apart from mentioned above, I get no other hdf5 file generated, can anyone tell me what's the problem?}
\paragraph{I had checked the address of my training dataset is right and the epoch is 50.}
\end{document}

**Comment by suikammd** [comment]
@anuragmishracse I‘m sorry to disturb you. After testing, the captions seem to be composed of similar fragments.

**Comment by Siri-NITK** [comment]
Can you please upload the weights-improvement-48.hdf5 file? The captions generated from the weights-improvement-04.gdf5 is not good.

**Comment by wrat** [comment]
I am also Facing same problem. After Training network , just generate 
weights-improvement-01.hdf5
weights-improvement-02.hdf5
weights-improvement-03.hdf5
weights-improvement-04.hdf5
weights-improvement-05.hdf5
not further improvement of weights. Can anyone share solution for same?
or Can you please upload the weights-improvement-48.hdf5 file

**Comment by suikammd** [comment]
Sorry, I only get the weights-improvement-8.hdf5 file and I cannot figure out why.

At2018-03-20 20:12:48，Abhishek Vermanotifications@github.comwrote:

I am also Facing same problem. After Training network , just generate
weights-improvement-01.hdf5
weights-improvement-02.hdf5
weights-improvement-03.hdf5
weights-improvement-04.hdf5
weights-improvement-05.hdf5
not further improvement of weights. Can anyone share solution for same?
or Can you please upload the weights-improvement-48.hdf5 file

**Comment by wrat** [comment]
@anuragmishracse  Can please see this issue ?

**Comment by sudhakar-sah** [comment]
@suikammd The reason may be because your loss is not reducing and it is not saving further weight files (we can set this in keras during training)

**Comment by ShixiangWan** [comment]
@sudhakar-sah After changing 1024 batch size, the loss of network is reducing. But the best weight is not good. For example:
![image](https://user-images.githubusercontent.com/9321757/40033473-2f34f4ee-582b-11e8-8c08-d979bde6929c.png)

**Comment by sudhakar-sah** [comment]
@ShixiangWan  I tried this repo few days ago and my loss was not reducing after 2-3 epochs. I am planning to work on this repo. There are few things that we can do for example, we can train a word2vec model on captions and initialize the embedding layer with these weights. are you still working on it ? if you like, we can work on it together ?

**Comment by Kinghup** [comment]
Do you solve this problem? I have same problem and the loss was not reducing. And only saving the model after 3 epochs. How should i do to reduce the loss?

**Comment by Kinghup** [comment]
> @ShixiangWan I tried this repo few days ago and my loss was not reducing after 2-3 epochs. I am planning to work on this repo. There are few things that we can do for example, we can train a word2vec model on captions and initialize the embedding layer with these weights. are you still working on it ? if you like, we can work on it together ?

Do you work with the repo now? I meet many problems , please help me.
